crates/ide-assists/src/handlers/raw_string.rs

```diff
@@ -156,11 +157,76 @@ pub(crate) fn remove_hash(acc: &mut Assists, ctx: &AssistContext<'_>) -> Option<
     })
 }
 
+// Assist: inline_str_literal
+//
+// Inline const variable as static str literal.
+//
+// ```
+// const STRING: &str = "Hello, World!";
+//
+// fn something() -> &'static str {
+//     STR$0ING
+// }
+// ```
+// ->
+// ```
+// const STRING: &str = "Hello, World!";
+//
+// fn something() -> &'static str {
+//     "Hello, World!"
+// }
+// ```
+pub(crate) fn inline_str_literal(acc: &mut Assists, ctx: &AssistContext<'_>) -> Option<()> {
+    let variable = ctx.find_node_at_offset::<ast::PathExpr>()?;
+
+    if let hir::PathResolution::Def(hir::ModuleDef::Const(konst)) =
+        ctx.sema.resolve_path(&variable.path()?)?
+    {
+        if !konst.ty(ctx.sema.db).as_reference()?.0.as_builtin()?.is_str() {
+            return None;
+        }
+
+        // FIXME: Make sure it's not possible to eval during diagnostic error
+        let value = match konst.value(ctx.sema.db)? {
+            ast::Expr::Literal(lit) => lit.to_string(),
+            ast::Expr::BlockExpr(_)
+            | ast::Expr::IfExpr(_)
+            | ast::Expr::MatchExpr(_)
+            | ast::Expr::CallExpr(_) => match konst.render_eval(ctx.sema.db) {
+                Ok(result) => result,
+                Err(_) => return None,
+            },
+            ast::Expr::MacroExpr(makro) => {
+                let makro_call = makro.syntax().children().find_map(ast::MacroCall::cast)?;
+                let makro_hir = ctx.sema.resolve_macro_call(&makro_call)?;
```

Comment:
Why not using `konst.render_eval` from beginning?

Reply:
During TDD, I ran into some evaluation errors for built-in macros, so I tried using the semantic expand method instead but got the same result. I now realize that the tests don't include code/std items, so it would have failed either way. 

So I guess I can change it back to using render_eval instead.